nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/heartbeat/AbstractHeartbeatMonitor.java

```diff
@@ -288,6 +292,11 @@ private void processHeartbeat(final NodeHeartbeat heartbeat) {
                 return;
             }
 
+            if (!isNodeApiReachable(nodeId)) {
+                logger.info("Received a connection request from {}, but the node's API address is not reachable, will not connect node to the cluster yet", nodeId);
```

Comment:
What do you think about adjusting the wording?
```suggestion
                logger.info("Node API Address [{}] not reachable: cluster connection request deferred pending successful network connection", nodeId);
```